src/Campaigns/resources/admin/components/CampaignFormModal/index.tsx

```diff
@@ -157,51 +154,7 @@ export default function CampaignFormModal({isOpen, handleClose, apiSettings, tit
         return null;
     };
 
-    const goalInputAttributes: {[selectedGoalType: string]: GoalInputAttributes} = {
-        amount: {
-            label: __('How much do you want to raise?', 'give'),
-            description: __('Set the target amount your campaign should raise.', 'give'),
-            placeholder: sprintf(__('eg. %s', 'give'),
-                currencyFormatter.format(2000),
-            ),
-        },
-        donations: {
-            label: __('How many donations do you need?', 'give'),
-            description: __('Set the target number of donations your campaign should bring in.', 'give'),
-            placeholder: __('eg. 100 donations', 'give'),
-        },
-        donors: {
-            label: __('How many donors do you need?', 'give'),
-            description: __('Set the target number of donors your campaign should bring in.', 'give'),
-            placeholder: __('eg. 100 donors', 'give'),
-        },
-        amountFromSubscriptions: {
-            label: __('How much do you want to raise?', 'give'),
-            description: __(
-                'Set the target recurring amount your campaign should raise. One-time donations do not count.',
-                'give'
-            ),
-             placeholder: sprintf(__('eg. %s', 'give'),
-                currencyFormatter.format(2000),
-            ),
-        },
-        subscriptions: {
-            label: __('How many recurring donations do you need?', 'give'),
-            description: __(
-                'Set the target number of recurring donations your campaign should bring in. One-time donations do not count.',
-                'give'
-            ),
-            placeholder: __('eg. 100 subscriptions', 'give'),
-        },
-        donorsFromSubscriptions: {
-            label: __('How many recurring donors do you need?', 'give'),
-            description: __(
-                'Set the target number of recurring donors your campaign should bring in. One-time donations do not count.',
-                'give'
-            ),
-            placeholder: __('eg. 100 subscribers', 'give'),
-        },
-    };
+    const goalInputAttribute = new CampaignGoalInputAttributes(selectedGoalType as GoalType, currency);
```

Comment:
Looks like we need to first make sure the goalType exists or else we get an error and the page doesn't load

<img width="1146" alt="Screenshot 2025-03-19 at 2 44 51 PM" src="https://github.com/user-attachments/assets/00d1f740-ce8c-4292-8be5-88a7c83c6105" />

Reply:
You must not face any issue with this: d2c8c12f63cd558d448802a2fe822c31ace1ca54

Reply:
Nice haha it also gives the modal a nice default as a bonus 🥇